src/ORM/Association.php

```diff
@@ -400,6 +401,17 @@ public function getTarget()
                 $config = ['className' => $this->_className];
             }
             $this->_targetTable = $tableLocator->get($registryAlias, $config);
+
+            $targetClassName = get_class($this->_targetTable);
```

Comment:
Final request, I promise...

Can we make this check only if the alias already exists in the locator? I'm trying to avoid the performance penalty of this check (probably really small anyways)